app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,58 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :find_post
```

Comment:
Don't you need to restrict find_post from being called on the index? It just throws an error when I try to view favourite posts.